modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/util/portlet_url/create_url.es.js

```diff
@@ -0,0 +1,124 @@
+import {isObject, isDef, isNull} from 'metal';
+
+export default function createURL(basePortletURL, parameters) {
+	let internalParameters = {};
```

Comment:
We can greatly simplify the algorithm:
1. Have `reservedParameters` as an array.
1. Create URLSearchParams from `basePortletURL`
1. Loop over `parameters` once. For each param:
    1. If the param is not reserved, check array from step 1, namespace it
    1. `set` param in `urlSearchParams`